plugin/pkg/admission/persistentvolume/label/admission.go

```diff
@@ -131,7 +137,11 @@ func (l *persistentVolumeLabel) getEBSVolumes() (aws.Volumes, error) {
 	defer l.mutex.Unlock()
 
 	if l.ebsVolumes == nil {
-		cloudProvider, err := cloudprovider.GetCloudProvider("aws", nil)
+		var cloudConfigReader io.Reader
+		if len(l.cloudConfig) > 0 {
+			cloudConfigReader = bytes.NewReader(l.cloudConfig)
+		}
+		cloudProvider, err := cloudprovider.GetCloudProvider("aws", cloudConfigReader)
```

Comment:
It is possible at least for kube-apiserver, since we already have a cloud object initialized there. It wouldn't work for federation api server though. 

What are we looking to gain by passing cloudprovider object instead of configuration bytes?